ambry-server/src/main/java/com/github/ambry/server/mysql/AccountReportsDao.java

```diff
@@ -0,0 +1,119 @@
+  private static final String insertSql =
+      String.format("INSERT INTO %s (%s, %s, %s, %s, %s, %s, %s) VALUES (?, ?, ?, ?, ?, ?, NOW())",
+          ACCOUNT_REPORTS_TABLE, CLUSTERNAME_COLUMN, HOSTNAME_COLUMN, PARTITION_ID_COLUMN, ACCOUNT_ID_COLUMN,
+          CONTAINER_ID_COLUMN, STORAGE_USAGE_COLUMN, UPDATED_AT_COLUMN);
+  private static final String querySqlForClusterAndHost =
+      String.format("SELECT %s, %s, %s, %s FROM %s WHERE %s = ? AND %s = ?", PARTITION_ID_COLUMN, ACCOUNT_ID_COLUMN,
+          CONTAINER_ID_COLUMN, STORAGE_USAGE_COLUMN, ACCOUNT_REPORTS_TABLE, CLUSTERNAME_COLUMN, HOSTNAME_COLUMN);
+  private final MySqlDataAccessor dataAccessor;
+  private final String clustername;
+  private final String hostname;
+
+  /**
+   * Constructor to create a {@link AccountReportsDao}.
+   * @param dataAccessor The underlying {@link MySqlDataAccessor}.
+   * @param clustername The name of the cluster this host is belonging to, like Ambry-prod, Ambry-video.
+   * @param hostname The name of the host. It should include the hostname and the port.
+   */
+  public AccountReportsDao(MySqlDataAccessor dataAccessor, String clustername, String hostname) {
+    this.dataAccessor = Objects.requireNonNull(dataAccessor, "MySqlDataAccessor is empty");
+    this.clustername = Objects.requireNonNull(clustername, "clustername is empty");
+    this.hostname = Objects.requireNonNull(hostname, "hostname is empty");
+  }
+
+  /**
+   * Update the storage usage for the given account/container.
+   * @param partitionId The partition id of this account/container usage.
+   * @param accountId The account id.
+   * @param containerId The container id.
+   * @param storageUsage The storage usage in bytes.
+   */
+  public void updateStorageUsage(short partitionId, short accountId, short containerId, long storageUsage)
+      throws SQLException {
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      PreparedStatement insertStatement = dataAccessor.getPreparedStatement(insertSql, true);
+      insertStatement.setString(1, clustername);
+      insertStatement.setString(2, hostname);
+      // The data type of partition id, account id and container id are not SMALLINT, but INT in MySQL, for
+      // future extension
+      insertStatement.setInt(3, partitionId);
+      insertStatement.setInt(4, accountId);
+      insertStatement.setInt(5, containerId);
+      insertStatement.setLong(6, storageUsage);
+      insertStatement.executeUpdate();
+      dataAccessor.onSuccess(Write, System.currentTimeMillis() - startTimeMs);
+    } catch (SQLException e) {
+      dataAccessor.onException(e, Write);
+      logger.error(String.format("Failed to execute updated on %s, with parameter %d %d %d %d", ACCOUNT_REPORTS_TABLE,
+          partitionId, accountId, containerId, storageUsage), e);
+      throw e;
+    }
+  }
+
+  public void queryForClusterAndHost(String clustername, String hostname, ContainerUsageFunction func)
```

Comment:
The reason to use cluster name is to do one query one given cluster name and get the stats for all hosts. It's not implemented yet, but it might be a good idea in the future.